### DISCUSSION TITLE
Feature Request: Allow per-monitor reserved space settings to prevent window overlap with dock areas

### DISCUSSION THREAD

**Discussion by samex** [velitasali/gnome-shell-extension-awesome-tiles#55] [Ideas]
**Description**: I’m loving the Awesome Tiles extension! However, I’d like to suggest a feature that would make it even more flexible and user-friendly for multi-monitor setups, especially those that use docks.

**Use Case**
I have a dock positioned at the bottom of one of my monitors, and currently, the tiled windows overlap with the dock, which interrupts workflow. I would like the ability to set reserved space individually for each monitor to ensure the tiled windows don’t cover the dock area or something else.

**Proposed Feature**
Could we add a per-monitor setting to define reserved space at specific edges (e.g., bottom, top, left, right) so that the Awesome Tiles extension respects these areas when tiling? Ideally, this could be implemented in the settings interface, allowing users to define edge-specific reserved spaces for each connected monitor.

**Benefits**
Enhanced usability: Ensures tiled windows don’t cover docks or panels on any monitor, making it easier to access these essential UI elements.
Improved flexibility: Allows users to customize the layout for multi-monitor setups, especially useful for those with asymmetric monitor arrangements or with different docks on each screen.
Seamless experience: Maintains visual harmony without unexpected window overlaps on specific monitors.

**Additional Considerations**
Possible default to detect common docks/panels automatically (if feasible).
Future expandability: A general framework for reserved areas might also help those who use custom panels or widgets.

Let me know if this feature aligns with the project’s roadmap! I’d be happy to assist with testing if needed.

**Comment by velitasali** [comment]
Hi @samex. Yes it is doable the way you described it, but I think instead of adding it like that I think it would make more sense to add more modes to the extension where we can have more vertical tiling options. That way, we can avoid introducing features that might clash with each other.
